src/state/achievements/helpers.ts

```diff
@@ -0,0 +1,52 @@
+import { getProfileContract } from 'utils/contractHelpers'
+import { campaignMap } from 'config/constants/campaigns'
+import { Achievement } from 'state/types'
+import { getAchievementTitle, getAchievementDescription } from 'utils/achievements'
+
+/**
+ * Gets all user point increase events on the profile filtered by wallet address
+ */
+export const getUserPointIncreaseEvents = async (account: string) => {
+  try {
+    const profileContract = getProfileContract()
+    const events = await profileContract.getPastEvents('UserPointIncrease', {
+      fromBlock: 'earliest',
+      toBlock: 'latest',
+      filter: {
+        userAddress: account,
+      },
+    })
+
+    return events
+  } catch (error) {
+    return []
+  }
+}
+
+/**
+ * Gets all user point increase events and adds achievement meta
+ */
+export const getAchievements = async (account: string): Promise<Achievement[]> => {
+  const pointIncreaseEvents = await getUserPointIncreaseEvents(account)
+
+  return pointIncreaseEvents.reduce((accum, event) => {
+    if (!campaignMap.has(event.returnValues.campaignId)) {
+      return accum
+    }
```

Comment:
Look here if we are missing achievements. Likely we did not connect it via the campaign ID properly.